privacyidea/api/lib/prepolicy.py

```diff
@@ -1941,3 +1941,34 @@ def _attestation_certificate_allowed(attestation_cert, allowed_certs_pols):
         else None
 
     return attestation_certificate_allowed(cert_info, allowed_certs_pols)
+
+
+def required_piv_attestation(request, action=None):
+    """
+    This is a token specific decorator certificate tokens for the endpoint
+    /token/init
+    According to the policy scope=SCOPE.ENROLL,
+    action=REQUIRE_ATTESTATION an exception is raised, if no attestation parameter is given.
+
+    It also checks the policy if the attestation should be verified and sets the
+    parameter verify_attestation accordingly.
+
+    :param request:
+    :param action:
+    :return:
+    """
+    from privacyidea.lib.tokens.certificatetoken import ACTION, REQUIRE_ACTIONS
+    ttype = request.all_data.get("type")
+    if ttype and ttype.lower() == "certificate":
+        # Get attestation certificate requirement
+        require_att = Match.user(g, scope=SCOPE.ENROLL, action=ACTION.REQUIRE_ATTESTATION,
```

Comment:
Also the `action` needs to prepend the token type (since this is a token specific action).

Reply:
I am not sure, if I understand this.
You mean the action needs to be prepended with the tokentype?

~~~~python
require_att = Match.user(g, scope=SCOPE.ENROLL, action="cert_{0!s}".format(ACTION.REQUIRE_ATTESTATION,
~~~~
??

Reply:
Yes, because it will be set in `get_dynamic_policy_definitions()` here:
https://github.com/privacyidea/privacyidea/blob/269d1ef24c7e1b3ddd204e5c84890bb7669a56a5/privacyidea/lib/token.py#L2462
before it is sent to the UI. So if the policy is set from the UI, the action is `"certificate_require_attestation"`.
And I would use something like:
```
... action="{0!s}_{1!s}".format(ttype, ACTION.REQUIRE_ATTESTATION), ...
```

Reply:
Thanks. Will do.

Reply:
Done in 47a374d3